sympy/functions/__init__.py

```diff
@@ -104,7 +104,7 @@
 
     'elliptic_k', 'elliptic_f', 'elliptic_e', 'elliptic_pi',
 
-    'beta',
+    'beta', 'betainc', 'regularized_betainc',
```

Comment:
I think it would be better if the name begins with beta somehow. I think Mathematica calls this `BetaRegularised` although I'm not sure if that's exactly the same function.

Reply:
Yes, that's the same function. Should I rename it to `beta_regularized`?

Reply:
Maybe it would be good to survey other languages, Matlab, mpmath, Maple, scipy etc.

Reply:
Matlab: `betainc`
mpmath: `betainc`
Maple: Couldn't find one
scipy: `betainc`
Mathematica: `BetaRegularized`
Octave: `betainc`
Maxima: `beta_incomplete_regularized`
So, which name would fit in best?

Reply:
Please suggest the name for the function. I think everything else is sorted here.

Reply:
Looks like `betainc` is the most common name and it's also consistent with both `scipy` and `mpmath` which sympy uses.

Do `scipy` and `mpmath` define the function in exactly the same way as sympy (for all possible complex values)?

Reply:
`mpmath` has the same definition of `betainc`. However, the same function is used for both the normal and the regularized incomplete beta function. This is generalized, which means the integral is computed from `x1 to x2`.

The definition of `scipy` is different. They define the regularized incomplete beta as `betainc` and the normal incomplete beta function can be obtained by multiplying with `beta(a, b)`. Further, the integral is always from 
`0 to x`.

Reply:
I'd suggest to go with betainc then but we will need to check that both mpmath and lambdify/scipy are working correctly.

Reply:
But we already have a class `betainc` for the normal incomplete beta function. Should I merge the `regularized_betainc` with `betainc`?

Reply:
We can't convert it to scipy because the integral is always from 0 to x, whereas we have x1 to x2 in our case. Conversion is only possible when x1=0.

Reply:
I don't know these functions well enough to suggest the right name.

It would be good to have names that are consistent with other libraries where possible. Having the same name but for something with a different definition is awkward though.

I think that `betainc_regularized` is better than `regularized_betainc`.

Reply:
Yes, I'll change it to `betainc_regularized`.